src/components/useExchangeCode.js

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import { useQuery } from 'react-query';
+import { useIntl } from 'react-intl';
+
+import useOkapiKy from '../useOkapiKy';
+import {
+  getLoginTenant,
+} from '../loginServices';
+import { useStripes } from '../StripesContext';
+
+const useExchangeCode = () => {
+  const stripes = useStripes();
+  const ky = useOkapiKy();
+  const intl = useIntl();
+
+  const [error, setError] = useState();
+
+  const urlParams = new URLSearchParams(window.location.search);
```

Comment:
I believe you are under react-router so you can safely use `useLocation`

Reply:
`useLocation` provides `{ hash, pathname, search, state }` but in order for the `redirect-uri` query-param to appease keycloak, it must be a fully-qualified URI, including protocol and host. 

Sure, I can pass value into the hook instead of accessing them there. It feels like six of one, half-dozen of another, but DI FTW, I guess?